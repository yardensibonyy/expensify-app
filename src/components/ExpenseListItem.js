import React from 'react';
import moment from 'moment';
import numeral from 'numeral';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <Link to={`/edit/${id}`} className="list-item">
    <div>
      <h3 className="list-item__title">{description}</h3>
      <span className="list-item__sub-title">{moment(createdAt).format('MMMM Do, YYYY')}</span>
    </div>
    <h3 className="list-item__date">{numeral(amount / 100).format('$0,0.00')}</h3>
  </Link>
);

export default ExpenseListItem;
