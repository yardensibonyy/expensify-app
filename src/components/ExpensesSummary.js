import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import expensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary = (props) => {
    return (
        <div>
            <div className='page-header'>
            <div className='content-container'>
                <h2>Expenses Summary</h2>
                <p>
                    Viewing {
                        props.expenses.length === 1 ? (
                            <span>
                                only one expense for
                                <span className='page-header__title'>
                                    {numeral(props.expenses[0].amount / 100).format(' $0,0.00')}
                                </span>
                            </span>
                        ) : (
                            <span>
                                <span className='page-header__title'>{props.expenses.length}</span> expenses totalling
                                <span className='page-header__title'>    
                                    {numeral(expensesTotal(props.expenses) / 100).format(' $0,0.00')}
                                </span> 
                            </span>
                        )
                    }
                </p>
                <div className='page-header__actions'>
                    <Link to='/create' className='button'>Add Expense</Link>
                </div> 
            </div>
            </div>
        </div>
    );
};

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }; 
};

export default connect(mapStateToProps)(ExpensesSummary);