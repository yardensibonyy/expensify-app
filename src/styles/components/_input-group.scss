.input-group {
    display: flex;
    margin-bottom: $m-size;
    flex-direction: column;
    @media (min-width: $desktop-breakpoint) { //for desktop
        margin-bottom: $l-size;
        flex-direction: row;   
    }
}

.input-group__item {
    margin-bottom: $s-size;
    @media (min-width: $desktop-breakpoint) { //for desktop
        margin: 0 $s-size 0 0;   
    }
}