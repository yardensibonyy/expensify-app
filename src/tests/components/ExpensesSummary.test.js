import React from 'react';
import { shallow, configure } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});