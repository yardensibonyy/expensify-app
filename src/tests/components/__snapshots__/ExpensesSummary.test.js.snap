// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h2>
        Expenses Summary
      </h2>
      <p>
        Viewing 
        <span>
          only one expense for
          <span
            className="page-header__title"
          >
             $1.95
          </span>
        </span>
      </p>
      <div
        className="page-header__actions"
      >
        <Link
          className="button"
          replace={false}
          to="/create"
        >
          Add Expense
        </Link>
      </div>
    </div>
  </div>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h2>
        Expenses Summary
      </h2>
      <p>
        Viewing 
        <span>
          <span
            className="page-header__title"
          >
            3
          </span>
           expenses totalling
          <span
            className="page-header__title"
          >
             $2,401.95
          </span>
        </span>
      </p>
      <div
        className="page-header__actions"
      >
        <Link
          className="button"
          replace={false}
          to="/create"
        >
          Add Expense
        </Link>
      </div>
    </div>
  </div>
</div>
`;
